src/cryptography/x509.py

```diff
@@ -1318,6 +1322,11 @@ def __init__(self, key_identifier, authority_cert_issuer,
         self._authority_cert_issuer = authority_cert_issuer
         self._authority_cert_serial_number = authority_cert_serial_number
 
+    @classmethod
+    def from_issuer_certificate(cls, certificate):
+        digest = _key_identifier_from_public_key(certificate.public_key())
+        return cls(digest, None, None)
```

Comment:
That makes the most sense to me.

On Sat, Aug 8, 2015 at 3:50 PM, Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/x509.py
> <https://github.com/pyca/cryptography/pull/2210#discussion_r36581178>:
>
> > @@ -1318,6 +1322,11 @@ def __init__(self, key_identifier, authority_cert_issuer,
> >          self._authority_cert_issuer = authority_cert_issuer
> >          self._authority_cert_serial_number = authority_cert_serial_number
> >
> > +    @classmethod
> > +    def from_issuer_certificate(cls, certificate):
> > +        digest = _key_identifier_from_public_key(certificate.public_key())
> > +        return cls(digest, None, None)
>
> Most implementations don't populate this, but it is possible to do so. I
> have no objection to changing it to from_public_key (and we can add a
> from_issuer_certificate later if people ask for it)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/2210/files#r36581178>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084